test/downstream-project-cmake-test/CMakeLists.txt

```
@@ -13,5 +13,26 @@ else()
   message(STATUS "The current exported EDM4HEP version is " ${EDM4HEP_VERSION})
 endif()
 
+# Use the EDM4HEP_DATA_DIR to derive extended data model
+message(STATUS "${EDM4HEP_DATA_DIR}")
+if ("${EDM4HEP_DATA_DIR}" STREQUAL "")
+  message(FATAL_ERROR "Unable to extend data model since no installed yaml file found.")
+else()
+  message(STATUS "Current EDM4HEP data model is available at ${EDM4HEP_DATA_DIR}.")
+  PODIO_GENERATE_DATAMODEL(edm4dis edm4dis.yaml headers sources
+    UPSTREAM_EDM edm4hep:${EDM4HEP_DATA_DIR}/edm4hep.yaml
+    OUTPUT_FOLDER ${CMAKE_CURRENT_BINARY_DIR}
+    IO_BACKEND_HANDLERS ${PODIO_IO_HANDLERS}
+  )
+
+  PODIO_ADD_DATAMODEL_CORE_LIB(edm4dis "${headers}" "${sources}"
+    OUTPUT_FOLDER ${CMAKE_CURRENT_BINARY_DIR}
+  )
+
```

```suggestion
target_link_libraries(edm4dis PUBLIC EDM4HEP::edm4hep)
```
The podio macro that builds the core lib does not know about the upstream model so it has to be explicitly linked against that.